snorkel/analysis/utils.py

```diff
@@ -12,21 +13,79 @@ def set_seed(seed: int) -> None:
     torch.manual_seed(seed)
 
 
-def probs_to_preds(probs: np.ndarray) -> np.ndarray:
+def _hash(i: int) -> int:
+    """Deterministic hash function."""
+    byte_string = str(i).encode("utf-8")
+    return int(hashlib.sha1(byte_string).hexdigest(), 16)
+
+
+def break_ties(
+    Y_prob: np.ndarray, tie_break_policy: Optional[str] = "random"
+) -> np.ndarray:
+    """Break ties among probabilistic labels according to given policy.
+
+    Policies to break ties include:
+    "abstain": return an abstain vote (0)
+    "true-random": randomly choose among the tied options
+    "random": randomly choose among tied option using deterministic hash
+
+    NOTE: if tie_break_policy="true-random", repeated runs may have slightly different results due to difference in broken ties
+
+    Parameters
+    ----------
+    Y_prob
+        An [n,k] array of probabilistic labels
+    tie_break_policy
+        Policy to break ties, by default 'random'
+
+    Returns
+    -------
+    np.ndarray
+        An [n] array of integer labels
+    """
+
+    n, k = Y_prob.shape
+    Y_pred = np.zeros(n)
+    diffs = np.abs(Y_prob - Y_prob.max(axis=1).reshape(-1, 1))
+
+    TOL = 1e-5
```

Comment:
nit: all-caps for scoped variable is odd, maybe make a function parameter with this as a default?